### PR TITLE
Trim spaces around k and v to tolerate extra whitespace in build.properties

### PR DESCRIPTION
**Pull request by invadergir**
We [noticed](https://github.com/sbt/sbt/issues/7587) what looked like a bash error when using a build.properties that looked like this:

```
sbt.version= 1.10.0
```

The error was:
```
/opt/sbt/bin/sbt: line 754: ((: 1.10.0 >= 2 : syntax error: invalid arithmetic operator (error token is ".10.0 >= 2 ")
/opt/sbt/bin/sbt: line 754: ((: 1.10.0 >= 1 : syntax error: invalid arithmetic operator (error token is ".10.0 >= 1 ")
false
```

I didn't realize `sbt` was a bash script, so I just added some whitespace trimming around the key and values inside `loadPropFile()`.  Now it is more forgiving of any spaces inserted around the '='.  All these test cases now work as expected with no errors:

```
sbt.version= 1.10.0
sbt.version =1.10.0
sbt.version = 1.10.0
```

I tested that spaces in front of the key are also trimmed as well, and of course if there is a '#' at the front (even with spaces in front of it), that is still ignored.

Hope this helps.

### CODE REVIEW

File: sbt
Comment (mkurz) [inline sbt]: Just wondering if it is possible to solve this with builtin bash functionality to avoid forking a new process. E.g. something like to replace all whitespaces https://stackoverflow.com/a/19661428/810109?

Comment (invadergir) [reply sbt]: Oh I just did it that way because I saw `sed` was used elsewhere.  Yes it looks like you can do this in plain bash.  I wasn't able to get the above example working (it seems to replace all spaces, not just beginning and ending), but this approach seems to work:  
```
str="  #  1.2.3  "
modStr="${str#"${str%%[![:space:]]*}"}"
modStr="${modStr%"${modStr##*[![:space:]]}"}"
echo "modStr is [$modStr]"
# prints '# 1.2.3'
```
That will work in POSIX compatible bashes.

Comment (invadergir) [reply sbt]: Ref: answer by Mateusz Piotrowski at [SOF](https://stackoverflow.com/questions/369758/how-to-trim-whitespace-from-a-bash-variable)

Comment (invadergir) [reply sbt]: I pushed a new version using a function derived from the above.

**Review by eed3si9n — APPROVED** [review]
Thanks both!

**Comment by eed3si9n** [comment]
@invadergir could you sign the Scala CLA please?

> Pull request submitted by invadergir
> CLA check for invadergir failed
> Please sign the Scala CLA to contribute to the Scala compiler.
> Go to https://www.lightbend.com/contribute/cla/scala and then
> comment on the pull request to ask for a new check.

**Comment by invadergir** [comment]
> @invadergir could you sign the Scala CLA please?
> 
> > Pull request submitted by invadergir
> > CLA check for invadergir failed
> > Please sign the Scala CLA to contribute to the Scala compiler.
> > Go to https://www.lightbend.com/contribute/cla/scala and then
> > comment on the pull request to ask for a new check.

So sorry, I'm waiting on approval from our lawyer.  I did this change while at work so they need to approve the CLA first.

**Comment by invadergir** [comment]
> > @invadergir could you sign the Scala CLA please?
> > > Pull request submitted by invadergir
> > > CLA check for invadergir failed
> > > Please sign the Scala CLA to contribute to the Scala compiler.
> > > Go to https://www.lightbend.com/contribute/cla/scala and then
> > > comment on the pull request to ask for a new check.
> 
> So sorry, I'm waiting on approval from our lawyer. I did this change while at work so they need to approve the CLA first.

This is done.  I guess the delay was because we were trying to get a corporate CLA and it seems this option doesn't exist anymore.  I got approval to sign as an individual though.

**Pull request merged**
